### ISSUE TITLE
508-defect-1 ❗ Launchblocker [KEYBOARD]: Copy link action must be accessible by keyboard using `enter` and `space`

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#23919]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh and Trevor

## User Story or Problem Statement

As a keyboard or screen reader user, I expect to be able to interact with buttons using the `space` and `enter` keys on my keyboard.

## Details

Keyboard and screen reader users will be unable to use the copy link button as both `space` and `enter` do not trigger copying of the link or confirmation of the event. Since the button does not have an `onclick` event handler, it will not be keyboard accessible. For more, read WCAG's [Making actions keyboard accessible by using the onclick event of anchors and buttons](https://www.w3.org/WAI/WCAG21/Techniques/client-side-script/SCR35).

My guess is that `onclick` may have been added to the `i` instead of the `button`, which would explain why the function doesn't work (the `i` has `hidden="true"`).

## Acceptance Criteria

- [ ] Copy link button is accessible by keyboard using `space` and `enter`

## Environment

`https://staging.va.gov/coronavirus-veteran-frequently-asked-questions/`

## Steps to Recreate

Tab to a copy link button and use the `space` and `enter` keys. Confirm no message appears and that the link is not copied.

## Proposed Solution

If possible, consider triggering the copy link action using the `onclick` event handler. This should enable access via the `keyboard` and `space` keys. [Making actions keyboard accessible by using the onclick event of anchors and buttons](https://www.w3.org/WAI/WCAG21/Techniques/client-side-script/SCR35).

**Comment by SKasimow** [comment]
@youngfreezyVA Hi Fareez - can you please review this bug and provide points and timing to correct? thx

cc @jenniferlee-dsva

**Comment by youngfreezyVA** [comment]
i'll have it in by EOW.

**Comment by youngfreezyVA** [comment]
just put in a PR. @joshkimux thank you for the excellent write up, it was the `onClick` event being on the icon instead of the button. that saved me debugging time!

 https://github.com/department-of-veterans-affairs/vets-website/pull/17182

**Comment by SKasimow** [comment]
@youngfreezyVA @joshkimux cheers to you both

**Comment by jenniferlee-dsva** [comment]
This is excellent feedback. Thank you @joshkimux.  Since this is severity level 1, let's resolve this first.

**Comment by joshkimux** [comment]
Confirmed that this is now working with both `space` and `enter` 🥳 

https://user-images.githubusercontent.com/14154792/118318624-a2567a00-b4c7-11eb-8cc9-a0e8adbca3af.mov
